src/Type/MediaItem/Mutation/MediaItemCreate.php

```diff
@@ -0,0 +1,221 @@
+					throw new \Exception( __( 'Sorry, you are not allowed to create mediaItems as this user', 'wp-graphql' ) );
+				}
+
+				/**
+				 * Set the file name, whether it's a local file or from a URL.
+				 * Then set the url for the uploaded file
+				 */
+				$file_name = basename( $input['filePath'] );
+				$uploaded_file_url = $input['filePath'];
+
+				/**
+				 * Require the file.php file from wp-admin. This file includes the
+				 * download_url and wp_handle_sideload methods
+				 */
+				require_once( ABSPATH . 'wp-admin/includes/file.php' );
+
+				/**
+				 * If the mediaItem file is from a local server, use wp_upload_bits before saving it to the uploads folder
+				 */
+				if ( 'file' === parse_url( $input['filePath'], PHP_URL_SCHEME) ) {
+					$uploaded_file = wp_upload_bits( $file_name, null, file_get_contents( $input['filePath'] ) );
+					$uploaded_file_url = $uploaded_file['url'];
+				}
+
+				/**
+				 * URL data for the mediaItem, timeout value is the default, see:
+				 * https://developer.wordpress.org/reference/functions/download_url/
+				 */
+				$timeout_seconds = 300;
+				$temp_file = download_url( $uploaded_file_url, $timeout_seconds );
+				/**
+				 * Handle the error from download_url if it occurs
+				 */
+				if ( is_wp_error( $temp_file ) ) {
+					throw new \Exception( __( 'Sorry, the URL for this file is invalid, it must be a valid URL', 'wp-graphql' ) );
+				}
+
+				/**
+				 * Build the file data for side loading
+				 */
+				$file_data = [
+					'name'     => $file_name,
+					'type'     => $input['fileType'],
+					'tmp_name' => $temp_file,
+					'error'    => 0,
+					'size'     => filesize( $temp_file ),
+				];
+
+				/**
+				 * Tells WordPress to not look for the POST form fields that would normally be present as
+				 * we downloaded the file from a remote server, so there will be no form fields
+				 * The default is true
+				 */
+				$overrides = [
+					'test_form' => false,
+				];
+
+				/**
+				 * Insert the mediaItem and retrieve it's data
+				 */
+				$file = wp_handle_sideload( $file_data, $overrides );
+				/**
+				 * Handle the error from wp_handle_sideload if it occurs
+				 */
+				if ( ! empty( $file['error'] ) ) {
+					throw new \Exception( __( 'Sorry, the URL for this file is invalid, it must be a path to the mediaItem file', 'wp-graphql' ) );
+				}
+
+				/**
+				 * Insert the mediaItem object and get the ID
+				 */
+				$media_item_args = MediaItemMutation::prepare_media_item( $input, $post_type_object, $mutation_name, $file );
+
+				/**
+				 * Get the post parent and if it's not set, set it to false
+				 */
+				$attachment_parent_id = ( ! empty( $media_item_args['post_parent'] ) ? $media_item_args['post_parent'] : false );
+
+				/**
+				 * Stop now if a user isn't allowed to edit the parent post
+				 */
+				// Attaching media to a post requires ability to edit said post.
```

Comment:
the mixed use of comment styles here is weird, and the second one is maybe slightly misplaced?